### PR TITLE
let cdn support sha256

### PR DESCRIPTION
**Pull request by zzy987**
let cdn support sha256
fix #517

### CODE REVIEW

File: cdnsystem/daemon/cdn/manager.go [hunk 1]
```diff
@@ -86,8 +90,27 @@ func (cm *Manager) TriggerCDN(ctx context.Context, task *types.SeedTask) (seedTa
...
+		switch digestType {
+		case digestutils.MD5:
+			fileDigest = md5.New()
+		case digestutils.SHA256:
+			fileDigest = sha256.New()
+		default:
+			digestType = digestutils.MD5
+			fileDigest = md5.New()
+		}
+	} else {
```
Comment (gaius-qi) [inline cdnsystem/daemon/cdn/manager.go:109]: why type is `MD5` when task.RequestDigest is empty.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by 244372610 — APPROVED** [review]

File: cdnsystem/daemon/cdn/manager.go (hunk 1 above)
Comment (zzy987) [reply cdnsystem/daemon/cdn/manager.go]: Ensure that the hash.Hash variable fileDigest is not nil when passed into the function detectCache

**Pull request merged**